### DISCUSSION TITLE
self hosted docker cannot upload

### DISCUSSION THREAD

**Discussion by plumber-craic** [ente-io/ente#2085] [Q&A] [answered]
I followed the docker guide and it seems to work. I can create an account, use cli to make it unlimited, and login with the android client. But uploading just has a spinner on the client, and the stdout of the dockers just shows this from museum:

`ente-museum-1           | INFO[4348]request_logger.go:83 func1 incoming                                      client_ip=x 
client_pkg=io.ente.photos client_version=0.8.131 query=count=18 req_body= req_id=0b20d588-041b-4253-afe6-638d80cc314f req_method=GET r
eq_uri=/files/upload-urls ua=Dalvik/2.1.0 (Linux; U; Android 14; SM-G998B Build/UP1A.231005.007)
ente-museum-1           | INFO[4348]file.go:283 GetUploadURLs Returning objectKeys: ...
ente-museum-1           | INFO[4348]request_logger.go:100 func1 outgoing                                      client_ip=x
client_pkg=io.ente.photos client_version=0.8.131 h_latency=43.786125ms latency_time=43.786125ms query=count=18 req_body= req_id=0b20d588-041b-4253-afe6-638d80cc314f req_method=GET req_uri=/files/upload-urls status_code=200 ua=Dalvik/2.1.0 (Linux; U; Android 14; SM-G
998B Build/UP1A.231005.007) user_id=x
`

I dont see an s3 403 error as mentioned in the faq and other discussion. Also logged into minio, created a token and secret, then used that to upload an image, which worked. 

When using the web client I get a CORS error - but it's really the android one I need to work. FYI for the non docker users: minio is the internal service name within the ente docker compose file and docker network.

```
[info] request failed - no response (put http://minio:3200/b2-eu-cen/x/2b8942fb-df28-44c6-bdab-cfbae49e3754?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20240610%2Feu-central-2%2Fs3%2Faws4_request&X-Amz-Date=20240610T172619Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=38f28aa97a1aaf4dc4f2a590e003383e70bc10ce624e7a198603fb39cb35ba26 log.ts:76:48
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://minio:3200/b2-eu-cen/x/2b8942fb-df28-44c6-bdab-cfbae49e3754?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=test%2F20240610%2Feu-central-2%2Fs3%2Faws4_request&X-Amz-Date=20240610T172619Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=38f28aa97a1aaf4dc4f2a590e003383e70bc10ce624e7a198603fb39cb35ba26. (Reason: CORS request did not succeed). Status code: (null).
```

Only thing added to museum.yml is making my user an admin. Docker is running on ubuntu 22 LTS.

**Comment by jonathanmeineke** [comment, marked as answer]
seems like its trying to use the docker network internal address in your browser. i had a similar issue.

i fixed it by changing the address in the config so the backend sends the right address for the Minio.
i'll have to check which one.

**Reply by plumber-craic** [reply]
thanks - i think you're right. After adding an entry to my hosts file for "minio" the web client upload worked, and I can see a successful PUT to http://minio:3200 in the browser. 

I don't see a way to override it in [local.yml](https://github.com/ente-io/ente/blob/main/server/configurations/local.yaml) - but maybe I misunderstand the s3 section. 

I notice there's a file called minio-provision.sh containing this, but changing the h0 alias doesnt seem to impact the connections i see in the web browser
```
#!/bin/sh

# Script used to prepare the minio instance that runs as part of the development
# Docker compose cluster.

while ! mc config host add h0 http://minio:3200 test testtest
do
   echo "waiting for minio..."
   sleep 0.5
done

cd /data

mc mb -p b2-eu-cen
mc mb -p wasabi-eu-central-2-v3
mc mb -p scw-eu-fr-v3
```

edit: it's [credentials.yml](https://help.ente.io/self-hosting/guides/configuring-s3). Will figure out how to enforce SSL and change the keys, since this is gonna be publicly available

**Reply by jonathanmeineke** [reply]
Probably a really bad way of doing it but i changed it in ./server/scripts/compose/credentials.yaml. as i don't really care about security as this is all behind a Tailscale Overlay Network.

**Reply by A4alli** [reply]
@jonathanmeineke I'm looking for setting ente photos server running via tailscale since year. Please do provide the compose.yml and everything.

**Reply by A4alli** [reply]
> Probably a really bad way of doing it but i changed it in ./server/scripts/compose/credentials.yaml. as i don't really care about security as this is all behind a Tailscale Overlay Network.

need the tailscale setup kindly

**Reply by jonathanmeineke** [reply]
@A4alli hey, sorry but i don't run this setup anymore and haven't for like at least half a year. i don't have the files anymore for this. sorry can't help you.

**Reply by A4alli** [reply]
thankyou for feedback. I just open a discussion **ente with tailscale**
